xdr/asset.go

```diff
@@ -431,3 +431,47 @@ func (a Asset) ToTrustLineAsset() TrustLineAsset {
 
 	return tla
 }
+
+func (a *Asset) GetCode() string {
+	switch a.Type {
+	case AssetTypeAssetTypeNative:
+		return ""
+	case AssetTypeAssetTypeCreditAlphanum4:
+		return string((*a.AlphaNum4).AssetCode[:])
+	case AssetTypeAssetTypeCreditAlphanum12:
+		return string((*a.AlphaNum12).AssetCode[:])
+	default:
+		return ""
+	}
+}
+
+func (a *Asset) GetIssuer() string {
+	switch a.Type {
+	case AssetTypeAssetTypeNative:
+		return ""
+	case AssetTypeAssetTypeCreditAlphanum4:
+		addr, _ := (*a.AlphaNum4).Issuer.GetAddress()
+		return addr
+	case AssetTypeAssetTypeCreditAlphanum12:
+		addr, _ := (*a.AlphaNum12).Issuer.GetAddress()
+		return addr
+	default:
+		return ""
+	}
+}
+
+func (a *Asset) LessThan(b Asset) bool {
+	if int32(a.Type) < int32(b.Type) {
+		return true
+	} else if int32(b.Type) < int32(a.Type) {
+		return false
+	}
+
+	if a.GetCode() < b.GetCode() {
+		return true
+	} else if b.GetCode() < a.GetCode() {
+		return false
+	}
+
+	return a.GetIssuer() < b.GetIssuer()
+}
```

Comment:
I've seen this logic a handful of times now, and each time I've had to triple-check it for correctness. Obviously, that's also what tests are for, but maybe there's a better way to code the logic. I think the following is equivalent:

```suggestion
func (a *Asset) LessThan(b Asset) bool {
	if int32(a.Type) == int32(b.Type) {
		if a.GetCode() == b.GetCode() {
			return a.GetIssuer() < b.GetIssuer()
		}
		return a.GetCode() < b.GetCode()
	}
	return int32(a.Type) < int32(b.Type)
}
```
Harder to understand, or easier?